src/shared/dissect-image.c

```diff
@@ -2061,6 +2070,33 @@ static int verity_partition(
 }
 #endif
 
+#if HAVE_LIBCRYPTSETUP
+static int dissected_partition_open_and_lock_decrypted(DissectedPartition *p, DissectImageFlags flags) {
+        _cleanup_close_ int fd = -1;
+        bool rw;
+
+        assert(p);
+
+        if (!p->decrypted_node)
+                return 0;
+
+        if (p->decrypted_node_fd >= 0)
+                return p->decrypted_node_fd;
+
+        rw = p->rw && !(flags & DISSECT_IMAGE_MOUNT_READ_ONLY);
+
+        fd = open(p->decrypted_node, O_NONBLOCK | O_CLOEXEC | O_NOCTTY | (rw ? O_RDWR : O_RDONLY));
```

Comment:
we can always open this in O_RDONLY mode. We just want to pin, not use for writes. And if we mount via /proc/self/fd/ then the kernel will only look at the inode we pin, and not what kind of fd we have...

Reply:
Is it possible to use O_PATH?

Reply:
I dont think so. I am pretty sure if you use O_PATH the bsd locks would end up being unaffected by the bsd locks on the non-O_PATH fds...

Reply:
OK. Thanks.